src/edu/ucsb/cs56/projects/games/subsink/Explosion.java

```diff
@@ -0,0 +1,56 @@
+package edu.ucsb.cs56.projects.games.subsink;
+
+import edu.ucsb.cs56.projects.games.subsink.Entity;
+
+import java.awt.*;
+import java.io.*;
+
+import edu.ucsb.cs56.projects.games.subsink.ImageLoader;
+import sun.audio.*;
+
+
+/**
+ * Created by andy on 2/25/18.
+ */
+public class Explosion {
+
+    public void playExplosionSound() throws IOException {
```

Comment:
`throws IOException` is bad. Catch the exception here instead of letting it propagate all the way up to `World`.

Reply:
and then remove all the silly try-catches you've added everywhere, and also the `throws IOException` on all the `update` methods. There's no reason any of those methods should throw anything.